crypto/threads_common.c

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+/**
+ * @file
+ * @brief Thread-local context-specific data management for OpenSSL
+ *
+ * This file implements a mechanism to store and retrieve context-specific
+ * data using OpenSSL's thread-local storage (TLS) system. It provides a way
+ * to associate and manage data based on a combination of a thread-local key
+ * and an `OSSL_LIB_CTX *` context.
+ *
+ * NOTE: This differs from the CRYPTO_THREAD_[get|set]_local api set in that
+ * this api stores a single OS level thread-local key per-thread, and manages
```

Comment:
It's not a single thread-local key per thread (the whole point is that there is one key, but each thread gets different local data) - its per process right?